coordinator/src/test/scala/com/socrata/datacoordinator/service/collocation/CoordinatedCollocatorTest.scala

```diff
@@ -628,7 +647,72 @@ class CoordinatedCollocatorTest extends FunSuite with Matchers with MockFactory
       result.moves.map(_.storeIdTo).toSet should be (Set(store1A, store3A))
     }
   )
-  
+
+  // Cost will be: (Cost(4, 40L, Some(10L)))
+  def testExpectRejectedExplainAndExecuteCollocation(message: String, costLimits: Cost, expectedRejectionMessage: String): Unit = {
+    // collocate: alpha.1 (store1A, store3A) with bravo.2 (store5A, store7A)
+    //
+    // moves:
+    //   bravo.2:   store5A, store7A -> store1A, store3A
+    //   charlie.3: store5A, store7A -> store1A, store3A
+    //
+    testExplainAndExecuteCollocation(
+      message,
+      { implicit coordinator =>
+        (coordinator.secondaryGroupConfigs _).expects().returns(Map(storeGroupA -> groupConfig(2, storesGroupA)))
+
+        expectSecondaryMoveJobsByDataset(storeGroupA, alpha1)
+        (coordinator.secondariesOfDataset _).expects(alpha1).once.returns(secondaries(store1A, store3A))
+
+        expectSecondaryMoveJobsByDataset(storeGroupA, bravo2)
+        (coordinator.secondariesOfDataset _).expects(bravo2).once.returns(secondaries(store5A, store7A))
+
+        expectCDOIAlpha1WithCollocationsComplex() // group for alpha.1
+        expectCDOIBravo2WithCollocationsSimple() // group for bravo.2
+
+      }, { implicit metric =>
+        expectDatasetMaxCostAlpha1WithCollocationsComplex
+        expectDatasetMaxCostBravo2WithCollocationsSimple
+
+      }) (
+
+      request(Seq((alpha1, bravo2))).copy(limits = costLimits),
+
+      { commonResult =>
+        val result = commonResult.right.get
+
+        result.cost should be (Cost(4, 40L, Some(10L)))
+
+        result.moves.size should be (4)
+        result.moves.map(_.datasetInternalName).toSet should be (Set(bravo2, charlie3))
+        result.moves.map(_.storeIdFrom).toSet should be (Set(store5A, store7A))
+        result.moves.map(_.storeIdTo).toSet should be (Set(store1A, store3A))
+      },
+
+      explainShouldBeRejected(expectedRejectionMessage),
+
+      executeShouldBeRejected(expectedRejectionMessage)
+    )
+  }
+
+  testExpectRejectedExplainAndExecuteCollocation(
+    "for collocation with moves exceeding the supplied move limit should reject the collocation",
+     costLimits.copy(moves = 3),
```

Comment:
It'd be a bit easier to read this if you just created a new costLimit instead of a copy with one param changed. Same for the 2 below.